File: transaction-service/src/main/java/ua/com/foxminded/restClient/service/TransactionService.java
package ua.com.foxminded.restClient.service;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import ua.com.foxminded.restClient.dto.TransactionDto;

import java.time.LocalDateTime;
import java.util.UUID;


public interface TransactionService {

    Page<TransactionDto> findAllByIdAndBetweenDate(UUID id, LocalDateTime start, LocalDateTime end, Pageable pageable);

}

File: transaction-service/src/main/kotlin/ua/com/foxminded/restClient/service/TransactionService.kt
package ua.com.foxminded.restClient.service

import java.time.LocalDateTime
import java.util.UUID
import org.springframework.data.domain.Pageable
import reactor.core.publisher.Flux
import ua.com.foxminded.restClient.dto.TransactionDto

interface TransactionService {
    fun findAllByIdAndBetweenDate(
        id: UUID,
        start: LocalDateTime?,
        end: LocalDateTime?,
        pageable: Pageable
    ): Flux<TransactionDto>

}
